### PR TITLE
Added error page on callback error

### PR DESCRIPTION
**Pull request by PaulAsjes**
Adds an error page when the AuthKit callback fails instead of sending back some generic JSON.

### CODE REVIEW

File: src/app/callback/error/page.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+"use client";
+
+import { Text, Heading, Flex, Box, Callout, Container } from "@radix-ui/themes";
+
+import { ExclamationTriangleIcon } from "@radix-ui/react-icons";
+
+export default function ErrorPage() {
+  return (
+    <>
```
Comment (kendallstrautman) [inline src/app/callback/error/page.tsx:9]:
```suggestion
```
Nit: extra fragment here

**Comment by kendallstrautman** [comment]
@PaulAsjes I made [some tweaks](https://github.com/workos/next-authkit-example/pull/11) in this PR.
- Refactored to a server component
- Adjusted some of the layout markup
- Moved the error page to `/error` since it makes more sense to me for only route handlers to be in the `callback` dir.

**Comment by kendallstrautman** [comment]
The only other thing to consider is whether we want to show the error message from [other errors](https://github.com/workos/next-authkit-example/blob/main/src/app/callback/route.ts#L50) beyond not getting `code`.

**Comment by PaulAsjes** [comment]
Merged in your changes and updated how we handle other auth errors, please take a look @kendallstrautman!

**Review by kendallstrautman — APPROVED** [review]
LGTM! 👌

**Pull request merged**